### PR TITLE
NCMC refactor -- cherry picking from hybrid-tests

### PR DESCRIPTION
**Pull request by juliebehr**
Extracting changes in the `hybrid-tests` branch that are independent to testing hybrid builder; here: refactoring `ncmc_switching.py` to inherit between `NCMCEngine` and `NCMCHybridEngine`

### CODE REVIEW

**Comment by jchodera** [comment]
Yay, travis is running again!

File: perses/annihilation/ncmc_switching.py [hunk 1]
```diff
@@ -538,52 +622,9 @@ def __init__(self, temperature=default_temperature, functions=None,
...
-        # Clean up context and integrator.
-        del context, integrator
-        # Return potential energy.
-        return -self.beta * potential
-
-    def _computeAlchemicalCorrection(self, unmodified_old_system,
-                                     unmodified_new_system, alchemical_system,
-                                     initial_positions, alchemical_positions,
-                                     final_hybrid_positions, final_positions,
+    def _computeAlchemicalCorrection(self, integrator, context,
```
Comment (jchodera) [inline perses/annihilation/ncmc_switching.py]: This `_computeAlchemicalCorrection` is different from the other one on line 161. Can they be combined, or do they need to be distinct?

Comment (juliebehr) [inline perses/annihilation/ncmc_switching.py]: They are completely different and need to be distinct

**Comment by juliebehr** [comment]
I think this is good to go -- the tests pass with this as-is, and there are other changes from `hybrid-tests` I'd like to pull in subsequently.

**Comment by jchodera** [comment]
Thanks!

**Pull request merged**